packages/firestore/src/util/sorted_set.ts

```diff
@@ -97,6 +97,19 @@ export class SortedSet<T> {
     }
   }
 
+  /** Finds the least element greater than `elem`. */
+  firstAfter(elem: T): T | null {
```

Comment:
So...  the consuming code is really just trying to iterate through the set.  But because we call firstAfter() every time we want to iterate to the next key, instead of it being an O(n) traversal it's an O(n log n) traversal...  which isn't really ideal.

So it seems like we should either try to expose an iterator interface of some kind on SortedSet or just take the lazy approach and call .toArray() (which creates an unnecessary copy but at least it's O(n)) and then iterate over that.